x/gamm/simulation/sim_msgs.go

```diff
@@ -140,18 +106,302 @@ func RandomCreateUniV2Msg(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context
 
 	// from the above selected account, determine the token type and respective weight needed to make the pool
 	for i := 0; i < len(poolCoins); i++ {
-		poolAssets = append(poolAssets, balancertypes.PoolAsset{
+		poolAssets = []balancertypes.PoolAsset{{
 			Weight: sdk.OneInt(),
 			Token:  poolCoins[i],
-		})
+		}}
 	}
+
 	return &balancertypes.MsgCreateBalancerPool{
 		Sender:     sender.Address.String(),
 		PoolParams: poolParams,
 		PoolAssets: poolAssets,
 	}, nil
 }
 
+// RandomSwapExactAmountIn utilizes a random pool and swaps and exact amount in for minimum of the secondary pool token
+// TODO: Improve this to swap through multiple pools
+func RandomSwapExactAmountIn(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context) (*types.MsgSwapExactAmountIn, error) {
+	// get random pool, randomly select one of the pool denoms to be the coinIn, other is coinOut
+	pool_id, pool, coinIn, coinOut, _, _, err := getRandPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// set the swap route to use this pool
+	route := []types.SwapAmountInRoute{{
+		PoolId:        pool_id,
+		TokenOutDenom: coinOut.Denom,
+	}}
+
+	// find an address that has a balance of the coinIn
+	sender, accCoinIn, senderExists := sim.SelAddrWithDenom(ctx, coinIn.Denom)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denom %s exists", coinIn.Denom)
+	}
+
+	// select a random amount that is upper-bound by the address's balance of coinIn
+	randomCoinSubset := sdk.NewCoins(sdk.NewCoin(accCoinIn.Denom, sim.RandomAmount(accCoinIn.Amount)))
+	if randomCoinSubset.IsZero() {
+		return nil, errors.New("cannot make a swap with zero amount")
+	}
```

Comment:
Can we make this condition impossible to occur? (By using a different amount selection function?)